src/test/java/org/opensearch/knn/plugin/action/RestKNNStatsHandlerIT.java

```diff
@@ -341,20 +342,23 @@ public void testScriptStats_multipleShards() throws Exception {
     public void testModelIndexHealthMetricsStats() throws Exception {
         // Create request that filters only model index
         String modelIndexStatusName = StatNames.MODEL_INDEX_STATUS.getName();
+        // index can be created in one of previous tests, and as we do not delete it each test the check below became optional
+        if (!systemIndexExists(MODEL_INDEX_NAME)) {
 
-        Response response = getKnnStats(Collections.emptyList(), Arrays.asList(modelIndexStatusName));
-        String responseBody = EntityUtils.toString(response.getEntity());
-        Map<String, Object> statsMap = createParser(XContentType.JSON.xContent(), responseBody).map();
+            final Response response = getKnnStats(Collections.emptyList(), Arrays.asList(modelIndexStatusName));
+            final String responseBody = EntityUtils.toString(response.getEntity());
+            final Map<String, Object> statsMap = createParser(XContentType.JSON.xContent(), responseBody).map();
 
-        // Check that model health status is null since model index is not created to system yet
-        assertNull(statsMap.get(StatNames.MODEL_INDEX_STATUS.getName()));
+            // Check that model health status is null since model index is not created to system yet
+            assertNull(statsMap.get(StatNames.MODEL_INDEX_STATUS.getName()));
+        }
 
         createModelSystemIndex();
```

Comment:
May be we should move this line of code into the if condition as we want to create it when it doesn't exist

Reply:
ack